src/game/GameHud.cpp

```diff
@@ -0,0 +1,466 @@
+    if (currentHP >= MAX_HEALTH) {
+        currentHP = MAX_HEALTH;
+    } else if (currentHP <= MIN_HEALTH) {
+        currentHP = MIN_HEALTH;
+    }
+
+    healthPercent = currentHP / MAX_HEALTH; //Percent of health left of the marine
+    size_t colorAmount = static_cast<size_t>(healthPercent * MAX_RGB_VALUE); //results in a value between 0 and 255
+
+    healthRGB[0] = MAX_RGB_VALUE - colorAmount; //set the Red value
+    healthRGB[1] = colorAmount; //Set the Green value
+    healthRGB[2] = MIN_RGB_VALUE; //Set the Blue value
+}
+
+/**
+ * Function: renderEquippedWeaponSlot
+ *
+ * Date:
+ * JF: March 27, 2017:
+ *
+ * Designer:
+ * Jacob Frank
+ *
+ * Programmer:
+ * Jacob Frank
+ *
+ *
+ * Interface: renderEquippedWeaponSlot(SDL_Rect screenRect)
+ *                  SDL_Rect screenRect: The Current screen properties (height, width)
+ *
+ * Notes:
+ * Function renders the equipped weapon slot to the screen.
+ * slot is positions in bottom right of screen next to the ammo clip.
+ */
+void GameHud::renderEquippedWeaponSlot(const SDL_Rect screenRect) {
+
+    //Makes the texture properties a squeare regardless of screen size
+    if (screenRect.w <= screenRect.h) {
+        equippedSlot.w = screenRect.w * EQUIPPED_SLOT_RAT;
+        equippedSlot.h = screenRect.w * EQUIPPED_SLOT_RAT;
+    } else {
+        equippedSlot.w = screenRect.h * EQUIPPED_SLOT_RAT;
+        equippedSlot.h = screenRect.h * EQUIPPED_SLOT_RAT;
+    }
+
+    //Set position to the left of the ammo clip
+    equippedSlot.x = ammoClipBackground.x - screenRect.w * PADDING_RAT - equippedSlot.w;
+
+    //align bottom of slot with other hud elements
+    equippedSlot.y = screenRect.h - screenRect.w * PADDING_RAT - equippedSlot.h;
+
+    Renderer::instance().render(equippedSlot, TEXTURES::EQUIPPED_WEAPON_SLOT);
+}
+
+/**
+ * Function: renderClip
+ *
+ * Date:
+ * JF: March 28, 2017:
+ *
+ * Designer:
+ * Jacob Frank
+ *
+ * Programmer:
+ * Jacob Frank
+ *
+ * Modified by:
+ * Jacob Frank (April 1, 2017)
+ *
+ * Interface: renderClip(SDL_Rect screenRect, Player p)
+ *                  SDL_Rect screenRect: The Current screen properties (height, width)
+ *                  Player p: The player in the game. used to find current equipped weapon
+ *
+ * Notes:
+ * Renders the weapon clip next to the currently equipped weapon.
+ * Visually displays how much ammo is left in the clip via a "depleting" image as bullets are fired.
+ * Ammo left in clip is calculated using the number of bullets in the clip and the max bullets the clip can hold.
+ *
+ * Revisions:
+ * JF April 1: Fixed Segfault bug resulting from not checking if the current weapon is a nullptr
+ */
+void GameHud::renderClip(const SDL_Rect screenRect, const Player p) {
+
+    float ammoInClip; //Ammo left in the weapons clip
+    float maxClipAmmo ; //Maximum bullets that can be held in the guns clip
+    float percentLeftinClip; //bullets in clip / max bullets of clip
```

Comment:
Temp variables should be declared as close to where they are used as possible
They should be made const if possible
I know for a fact that ammoInClip can have both done to it
I suggest reviewing usage of these temps and change the code accordingly